include/display/cfb.h

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2018 PHYTEC Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public Monochrome Character Framebuffer API
+ */
+
+#ifndef __CFB_H__
+#define __CFB_H__
+
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Display Drivers
+ * @defgroup display_interfaces Display Drivers
+ * @{
+ * @}
+ */
+
+/**
+ * @brief Public Monochrome Character Framebuffer API
+ * @defgroup monochrome_character_framebuffer Monochrome Character Framebuffer
```

Comment:
Could we have a general display API?

Displays could use different color space (B&W, RGB, YUV, etc), pixel and frame buffer formats (5-6-5 packed RGB, 4-2-0 planar YUV, etc.). We should be able to handle all of them using proposed API.

Reply:
>Could we have a general display API?

Yes we should, it is more WIP and I wanted to have some feedback.


>Displays could use different color space (B&W, RGB, YUV, etc)...

cfb should not support it, just monochrome and it should be as simple as possible.

Reply:
My point to have an generic framebuffer, which could be used to drive monochrome and color displays. IMHO having different API for monochrome and color is suboptimal.

Reply:
@pizi-nordic There is a pull request with a proposal for an API, #7833

Reply:
@vanwinkeljan: Thanks, I will look at it soon.